Kernel/Syscalls/uname.cpp

```diff
@@ -13,19 +16,28 @@ ErrorOr<FlatPtr> Process::sys$uname(Userspace<utsname*> user_buf)
     VERIFY_NO_PROCESS_BIG_LOCK(this);
     TRY(require_promise(Pledge::stdio));
 
-    utsname buf {};
-    memcpy(buf.sysname, "SerenityOS", 11);
-    memcpy(buf.release, "1.0-dev", 8);
-    memcpy(buf.version, "FIXME", 6);
+    utsname buf
+    {
+        "SerenityOS",
+            "", // Hostname, filled in below.
+            "", // "Release" (1.0-dev), filled in below.
+            "", // "Revision" (git commit hash), filled in below.
 #if ARCH(I386)
-    memcpy(buf.machine, "i686", 5);
+            "i686",
 #elif ARCH(X86_64)
-    memcpy(buf.machine, "x86_64", 7);
+            "x86_64",
 #elif ARCH(AARCH64)
-    memcpy(buf.machine, "AArch64", 7);
+            "AArch64",
 #else
 #    error Unknown architecture
 #endif
+    };
+
+    StringBuilder version_builder;
+    version_builder.appendff("{}.{}-dev", serenity_major_revision, serenity_minor_revision);
+    AK::TypedTransfer<u8>::copy(reinterpret_cast<u8*>(buf.release), version_builder.to_byte_buffer().data(), min(version_builder.length(), UTSNAME_ENTRY_LEN - 1));
+
+    AK::TypedTransfer<u8>::copy(reinterpret_cast<u8*>(buf.version), serenity_version.bytes().data(), min(serenity_version.length(), UTSNAME_ENTRY_LEN - 1));
```

Comment:
Lets do this properly and propagate errors, etc...
```suggestion
    KString version_string TRY(KString::formatted("{}.{}-dev", serenity_major_revision, serenity_minor_revision);
    AK::TypedTransfer<u8>::copy(buf.release, version_string.characters(), min(version_builder.length(), UTSNAME_ENTRY_LEN - 1));
    
    AK::TypedTransfer<u8>::copy(buf.version, serenity_version.characters_without_null_termination(), min(serenity_version.length(), UTSNAME_ENTRY_LEN - 1)); 
```